docs/api/QUIC_STREAM_EVENT.md

```diff
@@ -0,0 +1,273 @@
+} QUIC_STREAM_EVENT_TYPE;
+```
+
+The payload for QUIC stream events.
+
+```C
+typedef struct QUIC_STREAM_EVENT {
+    QUIC_STREAM_EVENT_TYPE Type;
+    union {
+        struct {
+            QUIC_STATUS Status;
+            QUIC_UINT62 ID;
+            BOOLEAN PeerAccepted : 1;
+            BOOLEAN RESERVED : 7;
+        } START_COMPLETE;
+        struct {
+            /* in */    uint64_t AbsoluteOffset;
+            /* inout */ uint64_t TotalBufferLength;
+            _Field_size_(BufferCount)
+            /* in */    const QUIC_BUFFER* Buffers;
+            _Field_range_(0, UINT32_MAX)
+            /* in */    uint32_t BufferCount;
+            /* in */    QUIC_RECEIVE_FLAGS Flags;
+        } RECEIVE;
+        struct {
+            BOOLEAN Canceled;
+            void* ClientContext;
+        } SEND_COMPLETE;
+        struct {
+            QUIC_UINT62 ErrorCode;
+        } PEER_SEND_ABORTED;
+        struct {
+            QUIC_UINT62 ErrorCode;
+        } PEER_RECEIVE_ABORTED;
+        struct {
+            BOOLEAN Graceful;
+        } SEND_SHUTDOWN_COMPLETE;
+        struct {
+            BOOLEAN ConnectionShutdown;
+            BOOLEAN AppCloseInProgress       : 1;
+            BOOLEAN ConnectionShutdownByApp  : 1;
+            BOOLEAN ConnectionClosedRemotely : 1;
+            BOOLEAN RESERVED                 : 5;
+            QUIC_UINT62 ConnectionErrorCode;
+            QUIC_STATUS ConnectionCloseStatus;
+        } SHUTDOWN_COMPLETE;
+        struct {
+            uint64_t ByteCount;
+        } IDEAL_SEND_BUFFER_SIZE;
+        struct {
+            /* out */ QUIC_UINT62 ErrorCode;
+        } CANCEL_ON_LOSS;
+    };
+} QUIC_STREAM_EVENT;
+```
+
+# Parameters
+
+`Type`
+
+The `QUIC_STREAM_EVENT_TYPE` that indicates which type of event this is, and which payload to reference (if any) for additional information.
+
+# Remarks
+
+## QUIC_STREAM_EVENT_START_COMPLETE
+
+This event is delivered when the [StreamStart](streamStart.md) operation completes. The accompanying payload contains data to indicate whether the operation succeeded or failed.
+
+### START_COMPLETE
+
+Additional state from the `StreamStart` operation is included in this payload struct/union.
+
+`Status`
+
+QUIC_STATUS value to indicate the operation completion code. Check for success using the QUIC_SUCCEEDED macro.
+
+`ID`
+
+Stream ID if available.
+
+`PeerAccepted`
+
+If TRUE, the peer has accepted the stream.
+
+## QUIC_STREAM_EVENT_RECEIVE
+
+Data received on an open stream is primarily delivered to the application through this event.
+
+### RECEIVE
+
+Received data on the stream is passed in this struct/union.
+
+`AbsoluteOffset`
+
+Absolute offset of the current data payload from the start of the receive operation.
+
+`TotalBufferLength`
+
+MsQuic indicates the total buffer length of the data in this parameter.
+
+See [Receiving Data On Streams](../Streams.md#Receiving) for further details on receiving data on a stream.
+
+Most notably, if the application...
```

Comment:
More broadly speaking, should we have these key points here, or should we just have a link from here to the comprehensive documentation on all the different receive patterns in Streams.md (above)?
The benefit to a link is having only one place to update.  The benefit to having key points here is that readers don't have to go to a new document.